src/components/ConfigEditor/DataSourceDescription.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import { css } from "@emotion/css";
+import { useTheme2 } from "@grafana/ui";
+
+type Props = {
+  dataSourceName: string;
+  docsLink: string;
+  hasRequiredFields?: boolean;
+};
+
+export const DataSourceDescription: React.FC<Props> = ({
```

Comment:
If we move it to grafana we will not be able to import it on grafana `8.4.7`, so we will not be able to use it in our plugins at the moment, right?

I think I can also add it to `@grafana/ui`, the same way as required label as you mentioned above. What do you think?

Reply:
Then it will have `FIelds marked as * are required` for any data source, but not all of them already have those fields marked. Also not all data sources have required fields maybe. So I guess we should leave it here for now.

Reply:
Yeah I guess something we can discuss. But if we merge it to core, clients running older versions of grafana won't see it even if the plugins will already be rewritten to support required fields

Reply:
@josiahg @szabalza what are your thoughts on this?

I don't think I can make the call by myself on where this should go but here are my opinions v

1. We may want to keep it here and not in grafana since there's many data sources that aren't getting overhauled atm and it'll be confusing for the inconsistency + required fields
2. If one of the mandates is to ensure consistency with older grafana versions - then having it only show up for new customers wouldn't work, right?

(Side note: I was actually just thinking about this problem, and how it will relate to some of the other components too. Let's discuss in the config redesign sync)

Reply:
I think the fact that it will appear only for the new grafana versions is not a big issue. This description is probably not critical. I am more concerned of it mentioning the * for required fields while they might not be there for the datasources without redesign.

Btw, do you know @yesoreyeram if it's possible to get the docs link for each datasource from grafana core? Or should we hardcode it if we decide to move it to core? The current component version has the link to the docs (visible on the screenshot above).

Reply:
> I think the fact that it will appear only for the new grafana versions is not a big issue. This description is probably not critical. I am more concerned of it mentioning the * for required fields while they might not be there for the datasources without redesign.

Yeah I agree. I think it'll be better for developers to manually update this instead of us pushing this to all the pages.

Reply:
One of the options is to have only the first part of the description without mentioning the information about the required fields. Then we can probably add it to grafana core.
And later we can add the information also about the asterisk. Or maybe we can add some flag to `plugin.json` and show the info about required fields only when the config page is redesigned. What do you think @yesoreyeram @catherineymgui ?

Reply:
Ok, I'll merge as is at the moment so that we can start using it already as changes to core and `plugin.json` might take some time and we are somewhat short on time.

I'll try to find time later and follow up on core changes, so that the description only renders for the plugins which have the docs link in `plugin.json` and also marked as "with redesigned config". We can later remove the usage of this component from the plugin itself together with changing it's `plugin.json`.